### ISSUE TITLE
Chiper / SSLv3 - Compare with some ISPs

### ISSUE THREAD

**Issue by aspdye** [mail-in-a-box/mailinabox#611]
Hi there,

I used "ssl-tools.net" for getting this information ;) This is the Report of an Mail-in-a-Box Setup (occams.info of @JoshData):

![occams info](https://cloud.githubusercontent.com/assets/12893462/11451517/3b2c9468-95c9-11e5-8952-1110d50de064.PNG)

SSL Tools has detected 2 main problems: SSLv3 and the weak chiper-suite. SSLv3 isn't a (big) problem for now. Everybody uses it. (except the guys of Mailbox.org) But nobody is using the weak chiper-suite (not even the providers who have no DANE :smile:). So why are we proving this with MiaB?

![1and1](https://cloud.githubusercontent.com/assets/12893462/11451534/a4880a3c-95c9-11e5-8ceb-8e4c26e04a8c.PNG)
(1&1) - No bad Chiper-Suite. No DANE TLS. SSLv3 supported.

![posteo](https://cloud.githubusercontent.com/assets/12893462/11451537/ae887148-95c9-11e5-8100-5c3d97e7fa82.PNG)
(Posteo) - No bad Chiper-Suite. SSLv3 supported.

![mailbox](https://cloud.githubusercontent.com/assets/12893462/11451538/b41cee0e-95c9-11e5-9859-c0d6f769ddcc.PNG)
(Mailbox.org) - Strict! No SSLv3 and no bad Chiper-Suite.

Conclusion: I think we should remove the "bad" Chiper-Suite ;) What do you think?

**Comment by MasterPCUK** [comment]
removing SSLv3 would be great +1

**Comment by JoshData** [comment]
See #371. It really just doesn't matter, and for a client old enough to _want_ the RC4 cipher, if the only alternative is dropping TLS entirely and communicating in the clear, well, that's not very helpful.

We'll do what Gmail does. Last we checked (in #371), Google was offering SSLv3. I don't recall if we checked if they offered the RC4 ciphers. Show me what they do and we'll do that.

**Comment by aspdye** [comment]
I have just tested it. (https://de.ssl-tools.net/mailservers/gmail.com) They are provinding SSLv3 aswell as the "ECDHE_RSA_WITH_RC4_128_SHA" chiper. But why are we doing it exactly as Gmail is doing? Other "big" Mailservers like Outlook.com are also not longer offering that Chiper. We could support SSLv3 (Outlook.com, ... does that aswell) but the chiper is flaged as "Critical" in the Report -> Thats the reason why i think we should fix it.

@JoshData What is the disadvantage of removing it?

**Comment by JoshData** [comment]
> But why are we doing it exactly as Gmail is doing? 

Because I trust Google's security people.

> What is the disadvantage of removing it?

Old mail servers that don't support newer ciphers will either not be able to send mail to the box, or they'll fall back to not using TLS at all.

**Comment by anoma** [comment]
Devils advocate. No one bans Google mails servers because they are universally trusted. Conversely every single advantage we can get in any compliance test makes it just that little bit less likely our small and unknown mail servers will be trusted.

Personally I would swap any tiny increase like this for talking to some archaic mail server that probably forgotten about and breached a decade ago any day of the week. In many ways there is an argument to make that we specifically dont want to talk to devices this old and at best we have anecdotal evidence it is even a problem.

My vote is we try to drop SSLv3 at least in a beta trial.

**Comment by aspdye** [comment]
Again: we Could (or should) keep sslv3. Everybody (nearly) does this. But Google and Yahoo are the only providers i Found Supporting this weak Chiper ;) I think the Chiper is Independent from the sslv3.

**Comment by anoma** [comment]
Yes sorry in my haste I was too generic in my wording. All SSL enabled services should get a high Qualsys score. Not just now but as we have seen in the past 18 months this will require ongoing changes.

**Comment by aspdye** [comment]
Is there any roadmap when Google is going to remove this Chiper Suite?

**Comment by aspdye** [comment]
The Mailcow Project seems to have solved this too ;)
https://ssl-tools.net/mailservers/hosted.mailcow.de

**Comment by JoshData** [comment]
Fyi, I just checked Google (aspmx.l.google.com) and they still offer SSLv3 on incoming mail (port 25):

```
  * SSLV3 Cipher Suites:
      Preferred:                       
                 RC4-SHA                       -              128 bits      250 2.0.0 OK uu6si9412596obc.50 - gsmtp
      Accepted:                        
                 AES128-SHA                    -              128 bits      250 2.0.0 OK w7si8639320obv.26 - gsmtp
                 RC4-MD5                       -              128 bits      250 2.0.0 OK w127si19104231oig.139 - gsmtp
                 DES-CBC3-SHA                  -              112 bits      250 2.0.0 OK t9si19092019oit.94 - gsmtp
                 RC4-SHA                       -              128 bits      250 2.0.0 OK o7si8493083obi.38 - gsmtp
                 AES256-SHA                    -              256 bits      250 2.0.0 OK j196si19097439oib.110 - gsmtp
```

In fact on port 587 too (smtp.gmail.com).

(using https://github.com/nabla-c0d3/sslyze)

**Comment by aspdye** [comment]
Can we remove the weak chipir suite? No provider (ok, gmail does :sweat_smile:) is offering this longer (GMX, 1&1, ...) and they are also great providers (from security-side :stuck_out_tongue_winking_eye:)

We would keep SSLv3 for some more time if you want :wink: 

I think there are enough examples of providers still using this :grimacing:

**Comment by yodax** [comment]
I might be wrong but; As long as smtp is vulnerable to downgrade attacks to plain text this doesn't matter. Mitm on smtp begins with plain text negotiations and removing starttls. Removing the cypher doesn't solve a problem. The problem is smtp.

**Comment by aspdye** [comment]
But why are all these providers have disabled this cypher? :grin: I think it does not hurt (and would make all the Checks of SSL-Tools green :sweat_smile:) 

@yodax is there any way to disable the possibility of downgrade attacks on SMTP?

**Comment by yodax** [comment]
As to why other providers do it, I can only guess. It might be to stop people asking to disable the cyphers. Seeing how much this comes up for a "small" project like mail-in-a-box.

To prevent downgrade to plain text you would have to disable plain text smtp like on the submission port. This will however break your ability to receive email from older clients or people from oppressed countries that actively downgrade mail traffic. 

I can't explain it as well as Filippo Valsorda, see his [blog](https://blog.filippo.io/the-sad-state-of-smtp-encryption/)

> STARTTLS has a glaring problem: it's negotiated over the plaintext channel.

> The S: 250 STARTTLS line is sent on the clear so an attacker performing a MitM attack can just block it, prevent it from ever reaching the client

Also look at what Thailand did: https://www.eff.org/deeplinks/2014/11/starttls-downgrade-attacks

I don't mind removing support for sslv3, just don't expect to be more secure from attackers. I think people who can do encryption, most likely can do better than sslv3. I'm guessing but I think clients fall roughly in 2 categories, modern and plain text. A relatively small group are older clients only supporting plain text and old cyphers. (none if this is backed by evidence)

Disclaimer: I am absolutely no expert on the matter, I just like reading about it. So I might be horribly wrong. Or just follow the general advice; don't trust strangers on the internet :smile:

If someone is more knowledgeable I would love to be corrected. 👍

**Comment by JoshData** [comment]
That's my understanding as well.

**Comment by JoshData** [comment]
https://discourse.mailinabox.email/t/unsecure-ciphers/1256

**Comment by JoshData** [comment]
Also further fixed in 6b73bb5d80fb5e8d42ba1921fdd383eafa168dd9, 5f5f00af4ac054d3b2ec51b38303ca5da73387e1.
